### PR TITLE
Fill in api/admin.go

### LINKED ISSUE #61: Admin API

### PR DESCRIPTION
**Pull request by Denton-L**
**Changelog:**

- Define api/admin.go (except for getAi)
- Fix API regex
- Define many to/from JSON functions

Close #61.

### CODE REVIEW

File: api/admin.go [hunk 1]
```diff
@@ -16,21 +18,79 @@ func InitAdmin() {
 }
 
 func getAllGames(s *Session, w http.ResponseWriter, r *http.Request) {
+	// params := mux.Vars(r)
+	// TODO: check the admin ID
```
Comment (DavidLu1997) [inline api/admin.go]: `s.IsAdmin()`

File: api/api.go [hunk 2]
```diff
@@ -36,7 +36,7 @@ func InitApi() {
 	BaseRoutes.Ai = BaseRoutes.Root.PathPrefix("/ai").Subrouter()
 	BaseRoutes.AiNeedGame = BaseRoutes.Ai.PathPrefix("/{game_id:[A-Za-z0-9]+}").Subrouter()
 
-	BaseRoutes.Admin = BaseRoutes.Root.PathPrefix("/admin/{admin_id:[A-za-z0-9+}").Subrouter()
+	BaseRoutes.Admin = BaseRoutes.Root.PathPrefix("/admin/{admin_id:[A-za-z0-9]+}").Subrouter()
```
Comment (DavidLu1997) [inline api/api.go]: On second thought, forget the admin id
`BaseRoutes.Admin = BaseRoutes.Root.PathPrefix("/admin").Subrouter()`

File: model/game.go [hunk 3]
```diff
@@ -47,15 +47,24 @@ func GameFromJson(data io.Reader) *Game {
...
 
 	return "[]"
 }
 
+func GameToJson(g *Game) string {
+	b, err := json.Marshal(g)
+	if err != nil {
+		return ""
+	} else {
+		return string(b)
```
Comment (DavidLu1997) [inline model/game.go]: No else-return

File: model/game_stats.go [hunk 4]
```diff
@@ -19,6 +20,35 @@ func (gs *GameStats) ToJson() string {
...
+
+	return "[]"
+}
+
+func GameStatsToJson(gs *Game) string {
+	b, err := json.Marshal(gs)
+	if err != nil {
+		return ""
+	} else {
+		return string(b)
```
Comment (DavidLu1997) [inline model/game_stats.go]: No else-return

File: model/player.go [hunk 5]
```diff
@@ -109,6 +109,24 @@ func PlayerFromJson(data io.Reader) *Player {
...
+
+	return "[]"
+}
+
+func PlayerToJson(p []*Player) string {
+	b, err := json.Marshal(p)
+	if err != nil {
+		return ""
+	} else {
+		return string(b)
```
Comment (DavidLu1997) [inline model/player.go]: No else-return

File: api/admin.go [hunk 6]
```diff
@@ -18,8 +19,13 @@ func InitAdmin() {
 }
 
 func getAllGames(s *Session, w http.ResponseWriter, r *http.Request) {
-	// params := mux.Vars(r)
-	// TODO: check the admin ID
+	params := mux.Vars(r)
+	if !s.IsAdmin() {
+		err := model.NewLocError("api/admin.go", "69", params, "You are not an admin!")
```
Comment (DavidLu1997) [inline api/admin.go]: `model.NewLocError("api/admin.go", "Not authorized", nil, "")`

**Comment by DavidLu1997** [comment]
+1

**Comment by Denton-L** [comment]
👍

**Pull request merged**